flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -420,89 +420,72 @@ protected int list(String[] args) {
 		}
 
 		try {
-			ActorGateway jobManagerGateway = getJobManagerGateway(options);
-
-			LOG.info("Connecting to JobManager to retrieve list of jobs");
-			Future<Object> response = jobManagerGateway.ask(
-				JobManagerMessages.getRequestRunningJobsStatus(),
-				clientTimeout);
+			CustomCommandLine<?> activeCommandLine = getActiveCustomCommandLine(options.getCommandLine());
+			ClusterClient client = activeCommandLine.retrieveCluster(options.getCommandLine(), config, configurationDirectory);
 
-			Object result;
+			Collection<JobDetails> jobDetails;
 			try {
-				result = Await.result(response, clientTimeout);
-			}
-			catch (Exception e) {
-				throw new Exception("Could not retrieve running jobs from the JobManager.", e);
+				CompletableFuture<Collection<JobDetails>> jobDetailsFuture = client.listJobs();
```

Comment:
Should we hide the fact that we are retrieving `JobDetails` internally and still only return a `Collection<JobStatus>`? The reason is that in the future we might change how we retrieve the job ids, job names plus their state.

Reply:
Even if we change how we retrieve it we can still wrap them in a JobDetails object; it doesn't necessarily leak to the outside how we retrieve them. Given that the CLI requires the job id, job name, start/end timestamp and job status we aren't saving much by creating a slimmed down JobDetails class.

Reply:
I thought about the `JobStatusMessage` class. I'm not entirely sure, because whatever you expose can be used and once we change that we no longer retrieve the full set of details for a job, it will be difficult to change it.

Reply:
I can change it to that.